### ISSUE TITLE
Browse a deleted version

### ISSUE THREAD

**Issue by kumar303** [mozilla/addons-code-manager#1162]
### Describe the problem and steps to reproduce it:

In the scenario described by https://github.com/mozilla/addons-code-manager/issues/1159 , the reviewer would like to browse a deleted version in Code Manager. This will help them look through historic code.

Actual example: https://reviewers.addons.mozilla.org/reviewers/review/989342/

### What happened?

The API returns a 404 (see https://github.com/mozilla/addons-server/issues/12725) and Code Manger displays an infinite loader.

### What did you expect to happen?

It should be possible to browse a deleted version.

### Anything else we should know?

The old File Viewer lets you browse a deleted version.

I'm not totally sure of the exact steps to set up the data that led to this because https://github.com/mozilla/addons-code-manager/issues/1130 suggests that browsing deleted versions might be possible in some scenarios.

**Comment by bobsilverberg** [comment]
This has been fixed. I'm not sure exactly when, but I believe it had to to with permissions.

**Comment by AlexandraMoga** [comment]
@bobsilverberg I dug more into this and it looks like we didn't reproduce the issue anymore because we were both logged in with admin users (at least in my case).

I've tried again to open the [deleted version](https://code.addons-dev.allizom.org/en-US/browse/618821/versions/1690254/) with a user that had `Addons:PostReview` and `Addons:ViewDeleted` permissions and I was able to reproduce the issue again.
The user was able to open the deleted version in the old file viewer.

So, if I'm correct, this issue is not fixed after all. I believe regular reviewers, not just admins, should have the option to see deleted versions.
